app/javascript/controllers/batch_operation_controller.ts

```diff
@@ -63,3 +96,44 @@ function switchButton(button: HTMLButtonElement, flag: boolean) {
     button.querySelectorAll('button').forEach((button) => disable(button));
   }
 }
+
+function displayNotice(inputs: HTMLInputElement[]) {
+  const checkbox_all = document.querySelector<HTMLInputElement>(
+    '#checkbox_all_batch_operation'
+  );
+  if (checkbox_all) {
+    if (checkbox_all.checked) {
+      show(document.querySelector('.fr-notice'));
+      hide(document.querySelector('#selected'));
+      show(document.querySelector('#not_selected'));
+    } else {
+      hide(document.querySelector('.fr-notice'));
+      deleteSelection();
+    }
+  }
+
+  const dynamic_number = document.querySelector('#dynamic_number');
+
+  if (dynamic_number) {
+    dynamic_number.textContent = inputs.length.toString();
+  }
+}
+
+function deleteSelection() {
+  const hidden_input_multiple_ids = document.querySelector<HTMLInputElement>(
+    '#input_multiple_ids_batch_operation'
+  );
+
+  if (hidden_input_multiple_ids) {
+    hidden_input_multiple_ids.value = '';
+  }
+
+  hide(document.querySelector('.fr-notice'));
+}
+
+function emptyCheckboxes() {
+  const inputs = document.querySelectorAll<HTMLInputElement>(
```

Comment:
marchait pas le this.inputTargets.forEach ?

Reply:
nop - j'ai un `this is undefined` - mais si tu as une façon de le faire mieux - je suis preneuse :)

Reply:
Tu peux déplacer cette function dans la class du controller et avoir `inputTargets`

Reply:
Le souci si j'utilise `inputTargets` c'est que ça ne décoche pas la checkbox "check_all"